<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>452</width>
    <height>565</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Debug devel server</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/images/network.png</normaloff>:/images/network.png</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <property name="margin">
     <number>2</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_8">
      <item>
       <widget class="QComboBox" name="comboBoxMode">
        <property name="toolTip">
         <string>Note: Under unix Local socket is Unix socket, under windows it's named pipe</string>
        </property>
        <item>
         <property name="text">
          <string>QTcpServer</string>
         </property>
         <property name="icon">
          <iconset resource="resources.qrc">
           <normaloff>:/images/network.png</normaloff>:/images/network.png</iconset>
         </property>
        </item>
        <item>
         <property name="text">
          <string>QLocalServer</string>
         </property>
         <property name="icon">
          <iconset resource="resources.qrc">
           <normaloff>:/images/pipe-socket.png</normaloff>:/images/pipe-socket.png</iconset>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_7">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_12">
      <item>
       <widget class="QCheckBox" name="compressedStream">
        <property name="text">
         <string>Compressed stream</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="compressionType">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <item>
         <property name="text">
          <string>Lz4hc</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>lz4</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>zlib</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>gzip</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="compressionOptions">
      <property name="title">
       <string>Compression options</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <property name="margin">
        <number>2</number>
       </property>
       <property name="spacing">
        <number>2</number>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label_compressio_buffer">
         <property name="text">
          <string>Compression buffer size:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="compressionBufferSize">
         <property name="minimum">
          <number>16</number>
         </property>
         <property name="maximum">
          <number>999999999</number>
         </property>
         <property name="value">
          <number>65536</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_compression_by_packet">
         <property name="text">
          <string>Compression by packet:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QCheckBox" name="compressionByPacket"/>
       </item>
      </layout>
      <zorder>label_compression_by_packet</zorder>
      <zorder>compressionBufferSize</zorder>
      <zorder>label_compressio_buffer</zorder>
      <zorder>compressionByPacket</zorder>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_7">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QToolButton" name="toolButton">
            <property name="maximumSize">
             <size>
              <width>22</width>
              <height>22</height>
             </size>
            </property>
            <property name="icon">
             <iconset resource="resources.qrc">
              <normaloff>:/images/list-remove-user.png</normaloff>:/images/list-remove-user.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QListWidget" name="listWidgetClient">
          <property name="maximumSize">
           <size>
            <width>200</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">background-position	: center;
background-repeat	: no-repeat;
background-color	: #fff;
background-attachment	: fixed;
background-image: url(:/images/system-users.png);</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="labelToClient">
            <property name="text">
             <string>To client:</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_5">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QToolButton" name="LoadFile">
            <property name="maximumSize">
             <size>
              <width>22</width>
              <height>22</height>
             </size>
            </property>
            <property name="icon">
             <iconset resource="resources.qrc">
              <normaloff>:/images/application-octet-stream.png</normaloff>:/images/application-octet-stream.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="ClearDataToClient">
            <property name="maximumSize">
             <size>
              <width>22</width>
              <height>22</height>
             </size>
            </property>
            <property name="icon">
             <iconset resource="resources.qrc">
              <normaloff>:/images/clear.png</normaloff>:/images/clear.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextEdit" name="textEditToClient">
          <property name="styleSheet">
           <string notr="true">background-image: url(:/images/upload.png);
background-position	: center;
background-repeat	: no-repeat;
background-color	: #fff;
background-attachment	: fixed;
</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QComboBox" name="comboBoxTypeTx">
            <item>
             <property name="text">
              <string>Direct</string>
             </property>
             <property name="icon">
              <iconset resource="resources.qrc">
               <normaloff>:/images/text.png</normaloff>:/images/text.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>toHex</string>
             </property>
             <property name="icon">
              <iconset resource="resources.qrc">
               <normaloff>:/images/hexa.png</normaloff>:/images/hexa.png</iconset>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="pushButtonSend">
            <property name="text">
             <string>Send</string>
            </property>
            <property name="icon">
             <iconset resource="resources.qrc">
              <normaloff>:/images/send.png</normaloff>:/images/send.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="labelFromClient">
            <property name="text">
             <string>From client:</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_6">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QToolButton" name="SaveFile">
            <property name="maximumSize">
             <size>
              <width>22</width>
              <height>22</height>
             </size>
            </property>
            <property name="icon">
             <iconset resource="resources.qrc">
              <normaloff>:/images/application-octet-stream.png</normaloff>:/images/application-octet-stream.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="ClearDataFromClient">
            <property name="maximumSize">
             <size>
              <width>22</width>
              <height>22</height>
             </size>
            </property>
            <property name="icon">
             <iconset resource="resources.qrc">
              <normaloff>:/images/clear.png</normaloff>:/images/clear.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextEdit" name="textEditFromClient">
          <property name="styleSheet">
           <string notr="true">background-position	: center;
background-repeat	: no-repeat;
background-color	: #eee;
background-attachment	: fixed;
background-image: url(:/images/download.png);</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QComboBox" name="comboBoxTypeRx">
            <item>
             <property name="text">
              <string>Direct</string>
             </property>
             <property name="icon">
              <iconset resource="resources.qrc">
               <normaloff>:/images/text.png</normaloff>:/images/text.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>toHex</string>
             </property>
             <property name="icon">
              <iconset resource="resources.qrc">
               <normaloff>:/images/hexa.png</normaloff>:/images/hexa.png</iconset>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QComboBox" name="comboBoxModeAppend">
            <item>
             <property name="text">
              <string>Update</string>
             </property>
             <property name="icon">
              <iconset resource="resources.qrc">
               <normaloff>:/images/overwirte.png</normaloff>:/images/overwirte.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Append</string>
             </property>
             <property name="icon">
              <iconset resource="resources.qrc">
               <normaloff>:/images/append.png</normaloff>:/images/append.png</iconset>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QGroupBox" name="groupBoxTcpSocket">
        <property name="title">
         <string>Tcp socket</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="labelPort">
           <property name="text">
            <string>Listen on this port:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="port">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
           <property name="value">
            <number>9999</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBoxLocalSocket">
        <property name="title">
         <string>Local socket</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="labelLocalSocket">
           <property name="text">
            <string>Listen on:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEditSocket">
           <property name="text">
            <string>socket-server</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_9">
      <item>
       <spacer name="horizontalSpacer_9">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonConnect">
        <property name="text">
         <string>Listen</string>
        </property>
        <property name="icon">
         <iconset resource="resources.qrc">
          <normaloff>:/images/connect.png</normaloff>:/images/connect.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_8">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBarApp"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>ClearDataToClient</sender>
   <signal>clicked()</signal>
   <receiver>textEditToClient</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>444</x>
     <y>175</y>
    </hint>
    <hint type="destinationlabel">
     <x>445</x>
     <y>256</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ClearDataFromClient</sender>
   <signal>clicked()</signal>
   <receiver>textEditFromClient</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>444</x>
     <y>314</y>
    </hint>
    <hint type="destinationlabel">
     <x>445</x>
     <y>395</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>compressedStream</sender>
   <signal>clicked(bool)</signal>
   <receiver>compressionType</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>119</x>
     <y>52</y>
    </hint>
    <hint type="destinationlabel">
     <x>287</x>
     <y>49</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
